public/lsp/dbo.php

```diff
@@ -859,7 +859,7 @@ function show_file($file_id, $user, $success = null) {
 			show_basic_file_info($object, false);
 			
 			// Bump the download button under details block
-			$url = htmlentities('download_file.php?file=' . $object['id'] . '&name=' . $object['filename']);
+			$url = 'download_file.php?file=' . $object['id'] . '&name=' . urlencode($object['filename']);
```

Comment:
This is only slightly better than the original.  It renames the file `Vall&amp;eacute;e des anciens V3.mp3`, which is still incorrect, but a bit better.

If I remove the `urlencode` altogether, I get `Vall&eacute;e des anciens V3.mp3`, which is still incorrect.

To fix it properly, we probably need to patch it here as well as here: https://github.com/LMMS/lmms.io/blob/6d0a4f113e31ef730006b0ec3bc7f4f18c7e6378/public/lsp/download_file.php#L23

Reply:
> This is only slightly better than the original. It renames the file `Vall&amp;eacute;e des anciens V3.mp3`, which is still incorrect, but a bit better.

The data in the database is html-entities escaped:

![image](https://user-images.githubusercontent.com/6829345/64970526-3a1f4300-d863-11e9-8879-48e92da7ad23.png)

So I can think of the following possibilities:

1. Un-escape the file name on the download link using HTML escaping and re-escape it using URL component escaping.
2. Un-escape the file name in `add_file.php` so that the database will record the "raw file name." This will only affect the newly uploaded assets, though.

Reply:
>To fix it properly, we probably need to patch it here too:

I have just tested:

```php
$url = 'download_file.php?file=' . $object['id'] . '&name=' . urlencode(html_entity_decode($object['filename']));
```

This will resolve to a correct file name, and adding a line above this... 

https://github.com/LMMS/lmms.io/blob/6d0a4f113e31ef730006b0ec3bc7f4f18c7e6378/public/lsp/download_file.php#L14-L20

... with

```php
$file_name = html_entity_decode($file_name);
```

which will produce the correct file name